Don't ignore robotNamespace

sfocke opened the pull request:
When creating the NodeHandle for reading the PID parameters, the model_nh was always ignored. Instead, all parameters were read from /gazebo_ros_control/pid_gains/<joint_name>/* instead of /<robot_name>/gazebo_ros_control/pid_gains/<joint_name>/*.

This commit restores the intended behavior, i.e., the parameters will now read from <robot_name>/..., where <robot_name> is specified via the robotNamespace plugin parameter or the parent name.

sfocke committed:
Don't ignore robotNamespace

When creating the NodeHandle for reading the PID parameters, the model_nh was always ignored. Instead, all parameters were read from /gazebo_ros_control/pid_gains/<joint_name>/* instead of /<robot_name>/gazebo_ros_control/pid_gains/<joint_name>/*.

This commit restores the intended behavior, i.e., the parameters will now read from <robot_name>/..., where <robot_name> is specified via the robotNamespace plugin parameter or the parent name.

sfocke commented:
Please note that this changes the current behavior to what was obviously intended by the author. A different possibility would be this:

```diff
-      const ros::NodeHandle nh(model_nh, "/gazebo_ros_control/pid_gains/" +
+      const ros::NodeHandle nh("/gazebo_ros_control/pid_gains/" +
```

... which doesn't change the behavior, only makes it clearer in the code what is actually happening.

j-rivero commented:
The change looks correct to me but we have to keep in mind the problem of current compatibility with code out there where people relies on the current behavior. I want to see the topics paths a contract like an API. In order to get this change into Kinetic, we should find a solution similar of what we did in https://github.com/ros-simulation/gazebo_ros_pkgs/pull/397 (legacy_mode parameter or similar).

sfocke committed:
Add legacyModeNS parameter

The legacyModeNS is only used when the DefaultRobotHWSim is used and does not affect the other RobotHWSims.

sfocke committed:
Enable warning when PID parameters were not found

sfocke commented:
I've added a legacyModeNS parameter to the plugin to deal with compatibility issues.

j-rivero commented:
@osrf-jenkins run tests please

j-rivero commented:
@osrf-jenkins run tests please

j-rivero commented:
@osrf-jenkins run tests please

kev-the-dev reviewed (approved):
Preserves current behavior by default and already merged into lunar/melodic

kev-the-dev commented:
@osrf-jenkins run tests please

j-rivero commented:
There is a failing test not related to this PR. Thanks @focke-85 for the contribution.
